### PR TITLE
domain,infoschema: put the @@tidb_schema_cache_size variable into effect

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: ref #50959 

Problem Summary:

### What changed and how does it work?

As the title says.
In the past, when we modify `@@global.tidb_schema_cache_size`,  nothing really happen.
Now we'll reset the cache capacity.
As experiments shows the current memory usage of the cache can meet our expection. 

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [X] Manual test (add detailed scripts or steps below)

With https://github.com/pingcap/tidb/pull/53301, the memory usage meets our expection.
The actual usage by schema cache is under control (but other components need more optimizations): 

B45rmzgq4A

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by ywqzzy — APPROVED** [review]

File: pkg/executor/show_stats_test.go [hunk 1]
```diff
@@ -292,7 +292,7 @@ func TestShowStatusSnapshot(t *testing.T) {
 	UPDATE variable_value = '%[2]s', comment = '%[3]s'`, safePointName, safePointValue, safePointComment)
 	tk.MustExec(updateSafePoint)
 
-	for _, cacheSize := range []int{1024, 0} {
+	for _, cacheSize := range []int{1024 * 1024 * 1024, 0} {
```
Comment (D3Hunter) [inline pkg/executor/show_stats_test.go:295]:
```suggestion
	for _, cacheSize := range []int{units.GiB, 0} {
```

File: pkg/executor/stale_txn_test.go [hunk 2]
```diff
@@ -893,7 +893,7 @@ func TestSetTransactionInfoSchema(t *testing.T) {
 	UPDATE variable_value = '%[2]s', comment = '%[3]s'`, safePointName, safePointValue, safePointComment)
 	tk.MustExec(updateSafePoint)
 
-	for _, cacheSize := range []int{1024, 0} {
+	for _, cacheSize := range []int{1024 * 1024 * 1024, 0} {
```
Comment (D3Hunter) [inline pkg/executor/stale_txn_test.go:896]: doitto

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**